R/checkmate.R

```diff
@@ -21,6 +21,107 @@
 #' @name assertions
 NULL
 
+# check equality ----
+
+#' Check if All Arguments Are Equal
+#'
+#' @description `r lifecycle::badge("experimental")`
+#' Elements of `...` must be numeric vectors or scalars.
+#'
+#' This function performs an element-by-element comparison of the first object
+#' provided in `...` with every other object in `...` and returns `TRUE` if all
+#' comparisons are equal within a given tolerance and `FALSE` otherwise.
+#'
+#' @param ... (`numeric`)\cr vectors to be compared
+#' @param tol (`numeric`)\cr the maximum difference to be tolerated when
+#' judging equality
+#'
+#' @note If there are any missing or infinite values in `...`, this function
+#'   returns `FALSE`, regardless of the values of other elements in `...`.
+#'
+#' @note If elements in `...` are not all of the same length, `FALSE` is returned.
+#'
+#' @return TRUE if all element-by-element differences are less than `tolerance`
+#' in magnitude, `FALSE` otherwise.
+#' @seealso [`assertions`] for more details.
+#'
+#' @export
+#' @examples
+#' check_equal(1:2, 1:2) # TRUE
+#' check_equal(1:2, 2:3) # "Not all equal"
+#' check_equal(Inf, Inf) # "Not all equal"
+#' check_equal(0.01, 0.02) # "Not all equal"
+#' check_equal(0.01, 0.02, tol = 0.05) # TRUE
+#' check_equal(1, c(1, 1)) # "Not all equal"
+check_equal <- function(..., tol = sqrt(.Machine$double.eps)) {
+  dot_args <- list(...)
+
+  sapply(dot_args, assert_numeric)
```

Comment:
better use `vapply` everywhere here (because it is safer)

Reply:
> better use `vapply` everywhere here (because it is safer)

`vapply` is new to me. Not sure I agree here.  

The potential problem is `vapply`'s `FUN.VALUE` argument.  From the online doc: "all values of FUN [should be] compatible with the FUN.VALUE, in that they must have the same length and type".  

The problem is that we don't know:

1. How many entries are in `...`
2. The length of each entry in `...`

and, in addition,

3. `assert_numeric` returns an object of the same type as its first argument (or throws an error)

I *think* we could work all that out, but it would add considerably to the complexity of the function.